File: ABC/061/d.py
import numpy as np
N, M = map(int, input().split())
mat = np.zeros((N, N))
for i in range(M):
    a, b, c = map(int, input().split())
    mat[a-1,b-1] = c
score = 0
tmp_v = 1
ever_v = [1]
while True:
    next_v = np.argmax(mat[tmp_v-1,:]) + 1
    score += mat[tmp_v-1,next_v-1]
    if next_v in ever_v:
        print('inf')
        break
    if next_v==N:
        print(score)
        break
    tmp_v = next_v

File: ABC/061/d.cpp
#include<iostream>
using namespace std;


int score(tmp_v, next_v){
    int nums[N+1];

    if(tmp_v==M) return score();
    if(tmpv_inBefore) return 0;
    for(int i=1; i<=N; i++){
        if(mat[tmp_v][i]>0){
            nums[i] = 1;
        }else{
            nums[i] = 0;
        }
    }
}

int main(){
    int N, M;
    cin >> N >> M;
    int mat[N+1][N+1];
    int memo[N+1][N+1];

    for(int i=0; i<=N; i++){
        for(int j=0; j<=N; j++){
            mat[i][j] = 0;
            memo[i][j] = 0;
        }
    }
    int a, b, c;
    for(int i=0; i<M; i++){
        cin >> a >> b >> c;
        mat[a][b] = c;
    }
    tmp_v = 1;
    while(true);
        for(int i=1; i<=N; i++){
            if(mat[tmp_v][i]>0){
                tmp_score += mat[tmp_v][i];
                tmp_v = i;
            }
        }


}
